app/web_framework/route_node.py

```diff
@@ -60,8 +61,26 @@ def add_children(self, child_nodes):
         self.children += child_nodes
         for node in child_nodes:
             node.parent = self
+            if version is not None:
+                node.version = version
+                node.assign_version_to_all_children(version)
         return self
 
+    def get_children(self, version: int):
+        """
+        Get all children routes that have the same version as requested.
+        :param version: The requested version.
+        """
+        return [child for child in self.children if child.version == version]
+
+    def assign_version_to_all_children(self, version: int):
+        """
+        Recursively assigns an API version to the current child and all of its direct children.
+        """
+        for node in self.children:
+            node.version = version
+            node.assign_version_to_all_children(version)
```

Comment:
Super minor, but I think if you make this 
```python
self.version = version
for node in self.children:
    node.assign_version_to_all_children(version)
```
then you won't need the `node.version = version` line 65 above. :neckbeard: